lib/hacluster.pm

```diff
@@ -306,7 +306,10 @@ Returns the number of nodes configured in the cluster.
 
 sub get_node_number {
     my $index = is_sle('15-sp2+') ? 2 : 1;
-    return script_output "crm_mon -1 | awk '/ nodes configured/ { print \$$index }'";
+    # Documentation of 'script_output' says "execution result may include extra serial output",
+    # in the test the output sometimes contains dmesg output, so we take only the last line.
+    my $out = script_output "crm_mon -1 | awk '/ nodes configured/ { print \$$index }'";
```

Comment:
LGTM, looking around in the same lib I see other functions takes different approach (I'm not suggesting you to change this PR, just collecting idea)

```
    my $crm_log_name = script_output("echo \"FILE=\|\$(ls $crm_log* | tail -1)\|\"");
    $crm_log_name =~ /FILE=\|([^\|]+)\|/;
    $crm_log_name = $1;
 ```

Reply:
`split /\n/, $out)[-1]` is not safe IMO.